### ISSUE TITLE
Indentation: Lambda in Enum indentation error loop

### ISSUE THREAD

**Issue by kapunga** [checkstyle/checkstyle#1141]
I'm creating an enum with reusable pieces of code and the indentation is giving me trouble.  We have Checkstyle configured to run as a git commit hook (as well as in out IDE). Here is an example code block for what I am trying to do:

```
package org.kapunga.weather;

/**
* @author kapunga
*/
public enum FloatMapper {
    IDENTITY("identity", o -> o.floatValue()),
    CEL_TO_FAREN("celToFaren", t -> {
        float f = t.floatValue();
        float i = f * (float) 1.8;

        return i + 32;
    }),
    FAREN_TO_CEL("farenToCel", t -> {
        float f = t.floatValue();
        float i = f - 32;

        return i / (float) 1.8;
    });

    public final String name;
    public final FloatMappingFunction func;

    FloatMapper(String name, FloatMappingFunction func) {
        this.name = name;
        this.func = func;
    }

    @FunctionalInterface
    public interface FloatMappingFunction {
        float map(Number n);
    }
}
```

When I try and commit this, I get checkstyle warnings:

> thor@ma-lt-thor ~/work/r9/trunk (client)$ git commit project/org/kapunga/weather/FloatMapper.java
> Java changes found, running Checkstyle.
> Checking source files for tabs.
> project/org/kapunga/weather/FloatMapper.java:9: warning: 'member def type' have incorrect indentation level 8, expected level should be 4.
> project/org/kapunga/weather/FloatMapper.java:10: warning: 'member def type' have incorrect indentation level 8, expected level should be 4.
> project/org/kapunga/weather/FloatMapper.java:15: warning: 'member def type' have incorrect indentation level 8, expected level should be 4.
> project/org/kapunga/weather/FloatMapper.java:16: warning: 'member def type' have incorrect indentation level 8, expected level should be 4.
> Commit aborted.

When I fix this to the following (Which I would argue is less readable):

```
package org.kapunga.weather;

/**
* @author kapunga
*/
public enum FloatMapper {
    IDENTITY("identity", o -> o.floatValue()),
    CEL_TO_FAREN("celToFaren", t -> {
    float f = t.floatValue();
    float i = f * (float) 1.8;

    return i + 32;
    }),
    FAREN_TO_CEL("farenToCel", t -> {
    float f = t.floatValue();
    float i = f - 32;

    return i / (float) 1.8;
    });

    public final String name;
    public final FloatMappingFunction func;

    FloatMapper(String name, FloatMappingFunction func) {
        this.name = name;
        this.func = func;
    }

    @FunctionalInterface
    public interface FloatMappingFunction {
        float map(Number n);
    }
}
```

I get errors as well:

> thor@ma-lt-thor ~/work/r9/trunk (client)$ git commit project/org/kapunga/weather/FloatMapper.java
> Java changes found, running Checkstyle.
> Checking source files for tabs.
> project/org/kapunga/weather/FloatMapper.java:9: warning: 'block' child have incorrect indentation level 4, expected level should be 8.
> project/org/kapunga/weather/FloatMapper.java:10: warning: 'block' child have incorrect indentation level 4, expected level should be 8.
> project/org/kapunga/weather/FloatMapper.java:12: warning: 'block' child have incorrect indentation level 4, expected level should be 8.
> project/org/kapunga/weather/FloatMapper.java:15: warning: 'block' child have incorrect indentation level 4, expected level should be 8.
> project/org/kapunga/weather/FloatMapper.java:16: warning: 'block' child have incorrect indentation level 4, expected level should be 8.
> project/org/kapunga/weather/FloatMapper.java:18: warning: 'block' child have incorrect indentation level 4, expected level should be 8.
> Commit aborted.

So as you can see, I can't get this to agree on a correct indentation level.  I think the first example should be correct, the second example makes it hard to see where each enum begins.  When I do something like this:

```
FloatMappingFunction celToFaren = t -> {
    float f = t.floatValue();
    float i = f * (float) 1.8;

    return i + 32;
};
```

I have no checkstyle errors.
##

**Comment by plan3d** [comment]
This issue seems fixed in checkstyle 6.11.1. Output with actual version 8.41:

`$ cat test.java 
package org.kapunga.weather;

/**
* @author kapunga
*/
public enum FloatMapper {
    IDENTITY("identity", o -> o.floatValue()),
    CEL_TO_FAREN("celToFaren", t -> {
        float f = t.floatValue();
        float i = f * (float) 1.8;

        return i + 32;
    }),
    FAREN_TO_CEL("farenToCel", t -> {
        float f = t.floatValue();
        float i = f - 32;

        return i / (float) 1.8;
    });

    public final String name;
    public final FloatMappingFunction func;

    FloatMapper(String name, FloatMappingFunction func) {
        this.name = name;
        this.func = func;
    }

    @FunctionalInterface
    public interface FloatMappingFunction {
        float map(Number n);
    }
}

$ cat checkstyle_config_test.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name = "Checker">
    <module name="TreeWalker">
        <module name="Indentation">
            <property name="lineWrappingIndentation" value="4"/>
        </module>
    </module>
</module>

$ java -jar ./checkstyle-8.41-all.jar -c ./checkstyle_config_test.xml ./test.java
Starting audit...
Audit done.`

Version 6.11 produces:

`Starting audit...
./test.java:9: error: 'member def type' have incorrect indentation level 8, expected level should be 4.
./test.java:10: error: 'member def type' have incorrect indentation level 8, expected level should be 4.
./test.java:15: error: 'member def type' have incorrect indentation level 8, expected level should be 4.
./test.java:16: error: 'member def type' have incorrect indentation level 8, expected level should be 4.
Audit done.
Checkstyle ends with 4 errors.`

**Comment by strkkk** [comment]
@kapunga please upgrade your checkstyle version.
@plan3d thatnks for checking.

**Comment by pavelicii** [comment]
Having same error loop in 9.2 with plain `<module name="Indentation"/>`  when each enum argument is on new line.

Example:
```java
package some.pckg;

import java.util.function.Function;

public enum SomeEnum {

    ENUM_VALUE(
            v -> v, // violation (line 13)
            new Object()
    );

    private final Function<Object, Object> function;
    private final Object object;

    SomeEnum(Function<Object, Object> function, Object object) {
        this.function = function;
        this.object = object;
    }
}

```

Error:
```
[ant:checkstyle] [ERROR] ...\SomeEnum.java:8:13: 'lambda arguments' has incorrect indentation level 12, expected level should be 16. 
[Indentation]
```

**Comment by romani** [comment]
@pavelicii, please open new issue
